ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Protocol/Praos.hs

```diff
@@ -46,6 +46,45 @@ type instance ShelleyProtocolHeader (Praos c) = Header c
 
 data PraosEnvelopeError
   = ObsoleteNode Version Version
+    -- ^ This is a subtle case.
+    --
+    -- This node is explicitly rejecting the header, but the header isn't
+    -- necessarily _directly_ at fault.
+    --
+    -- This rejection specifically happens when the ledger state being used to
+    -- validate this header contains a protocol major version (the first
+    -- 'Version') that exceeds the maximum major protocol version allowed for
+    -- this era this specific node's configuration (the second 'Version'). The
+    -- only thing the header did "wrong" was extend such a ledger state.
```

Comment:
Maybe say "*ticked* ledger state" as in the unticked one, the major protocol version was not yet incremented.

Reply:
Done, thanks!